stan/math/opencl/kernel_generator/multi_result_kernel.hpp

```diff
@@ -321,6 +321,16 @@ class results_cl {
           std::get<Is>(results_), std::get<Is>(exprs.expressions_))...));
     });
   }
+  template <typename... T_expressions,
+            typename = std::enable_if_t<sizeof...(T_results)
+                                        == sizeof...(T_expressions)>>
+  void operator=(expressions_cl<T_expressions...>&& exprs) {
+    index_apply<sizeof...(T_expressions)>([this, &exprs](auto... Is) mutable {
+      assignment_impl(std::tuple_cat(make_assignment_pair(
+          std::get<Is>(results_),
+          std::forward<T_expressions>(std::get<Is>(exprs.expressions_)))...));
+    });
+  }
 
   /**
    * Incrementing \c results_ object by \c expressions_cl object
```

Comment:
Missing docs for the templates. Those are pretty complicated requires so please write out what's going on there

Reply:
Template is doced on line 339. The require is also relatively simple - just the number of results and expressions must match.